third_party/cached_httpfs/httpfs.cpp

```diff
@@ -323,17 +334,17 @@ unique_ptr<ResponseWrapper> HTTPFileSystem::GetRequest(FileHandle &handle, strin
 				    hfh.state->total_bytes_received += data_length;
 			    }
 			    if (!hfh.cached_file_handle->GetCapacity()) {
-				    hfh.cached_file_handle->AllocateBuffer(data_length);
+				    hfh.cached_file_handle->Allocate(data_length);
```

Comment:
Are we making sure that these are initialized files earlier on?
Since CachedFile::Allocate will throw an InternalException if `initialized` is set, this should be verified elsewhere or be downgraded to a regular exception as InternalExceptions are unrecoverable

Reply:
Most of this code around file read/write/reserve is copied from in-memory cache which need to check conditions but in our case (cache file is not available until completely downloaded). Methods that are used in file caching are used only in GET request which is issued once we have WRITE_LOCK and opened file handle. Changed all this check to be ASSERT. Comment applies to next review change request.